intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/base/problems/InvalidDescriptorErrors.kt

```diff
@@ -92,4 +93,20 @@ class ContainsNewlines(propertyName: String, descriptorPath: String? = null) : I
   override val detailedMessage = "<$propertyName> can't contain newlines"
 
   override val level = Level.ERROR
-}
\ No newline at end of file
+}
+
+class VendorCannotBeEmpty(descriptorPath: String? = null
+) : PropertyNotSpecified("vendor", descriptorPath) {
+
+  private val solutionHint = ProblemSolutionHint(
+          """<vendor email="joe@example.com">Joe Doe</vendor>""",
+          URL("https://plugins.jetbrains.com/docs/intellij/plugin-configuration-file.html#idea-plugin__vendor")
```

Comment:
Where will we use the url?

Reply:
This is a very first effort to improve messages on errors and warnings. It will be improved on further PRs and integrated with the rest of the ecosystem.